Dockerfile.dnsmasq-nanny

```diff
@@ -11,10 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM ARG_REGISTRY/k8s-dns-dnsmasq-ARG_ARCH:ARG_VERSION
+
+ARG registry=staging-k8s.gcr.io
+ARG arch=amd64
+ARG version
+
+FROM ${registry}/k8s-dns-dnsmasq-${arch}:${version}
+
+# Needs to be redefined in order to make this variable present during build
+ARG arch=amd64
```

Comment:
How would this now build for other ARCH that amd64? Previously we were passing in ARG_ARCH for each defined ARCH from rules.mk.